hw1/02-palindrome.js

```diff
@@ -1,3 +1,50 @@
-const elem = document.querySelector('input');
+const elem = document.querySelector("input");
+const palindromeText = document.getElementById("palindrome");
 
-elem.addEventListener('input', handleInput);
+const isPalindrome = (num) => {
+  let l = 0;
+  let r = num.length - 1;
+
+  while (l <= r) {
+    if (num[l] !== num[r]) {
+      return false;
+    }
+    l += 1;
+    r -= 1;
+  }
+
+  return true;
+};
+
+const handleInput = () => {
+  const { value } = elem;
+  palindromeText.innerText = "";
+  palindromeText.classList.remove("text-success", "text-danger");
+
+  if (value.length === 0) {
+    palindromeText.innerText = "";
+    return;
+  }
+
+  if (Number.isNaN(value) || value.includes(" ")) {
+    palindromeText.innerText = "Non-numeric characters detected.";
+    palindromeText.classList.add("text-danger");
+    return;
+  }
```

Comment:
I believe that defining the input type as a number should generally prevent text input. The field format itself wouldn't allow a text input when the input type is a number.  So in my view, additional checks may not be necessary.